src/infrastructure/utils/credentials.utils.ts

```diff
@@ -75,16 +39,31 @@ export default class CredentialsUtils {
     return token.split("|")[1];
   }
 
-  public static createToken(sigaBearerToken: string, miUtemCookies: Cookie[]): string {
-    let miUtemToken = "";
-    if (miUtemCookies != null) {
-      let sessionId: string = miUtemCookies.find(cookie => cookie.name == "sessionid")?.value || "";
-      let csrfToken: string = miUtemCookies.find(cookie => cookie.name == "csrftoken")?.value || "";
+  public static getAcademiaCookies(token: string): Cookie[] {
+    const base64 = token.split("|")[2]
+    if(!CredentialsUtils.isBase64(base64)) {
+      return [];
+    }
+
+    const data: string[] = atob(base64).split("|"); // Formato: expira|cookie1|cookie2|cookie3...
+    if(data.length <= 1) { // Valida el formato
+      return [];
+    }
 
-      miUtemToken = sessionId + csrfToken;
+    const expira = new Date(parseInt(data[0]) * 1000);
+    if(expira < new Date()) { // Valida que no haya expirado
+      return [];
     }
 
-    return miUtemToken + "|" + sigaBearerToken;
+    return data.slice(1).map(it => Cookie.parse(atob(it)));
+  }
+
+  public static createToken(sigaBearerToken: string, miUtemCookies: Cookie[], academiaCookies: Cookie[]): string {
+    const expira = new Date((new Date).getTime() + (6 * 60 * 60 * 1000));
+    const miutemToken = btoa(`${expira}|${miUtemCookies.map(it => btoa(it.raw)).join('|')}`)
+    const academiaToken = btoa(`${expira}|${academiaCookies.map(it => btoa(it.raw)).join('|')}`)
```

Comment:
Cuando corro la función me arroja el error `ReferenceError: btoa is not defined`